pkg/scheduler/algorithm/priorities/requested_to_capacity_ratio.go

```diff
@@ -84,35 +86,64 @@ func NewFunctionShape(points []FunctionShapePoint) (FunctionShape, error) {
 	return pointsCopy, nil
 }
 
+// NewResources creates instance of Resources in a safe way performing all
+// necessary sanity checks.
+func NewResources(resources Resources) (Resources, error) {
+	n := len(resources)
+
+	if n == 0 {
+		resources = Resources{v1.ResourceMemory: 1, v1.ResourceCPU: 1}
+	}
+
+	for resource, weight := range resources {
+		if weight < 0 {
+			return nil, fmt.Errorf("resource %s weight %d must not be less than 0", string(resource), weight)
+		}
+	}
+	return resources, nil
+}
+
 // RequestedToCapacityRatioResourceAllocationPriorityDefault creates a requestedToCapacity based
 // ResourceAllocationPriority using default resource scoring function shape.
 // The default function assigns 1.0 to resource when all capacity is available
 // and 0.0 when requested amount is equal to capacity.
 func RequestedToCapacityRatioResourceAllocationPriorityDefault() *ResourceAllocationPriority {
-	return RequestedToCapacityRatioResourceAllocationPriority(defaultFunctionShape)
+	return RequestedToCapacityRatioResourceAllocationPriority(defaultFunctionShape, defaultResources)
 }
 
 // RequestedToCapacityRatioResourceAllocationPriority creates a requestedToCapacity based
 // ResourceAllocationPriority using provided resource scoring function shape.
-func RequestedToCapacityRatioResourceAllocationPriority(scoringFunctionShape FunctionShape) *ResourceAllocationPriority {
-	return &ResourceAllocationPriority{"RequestedToCapacityRatioResourceAllocationPriority", buildRequestedToCapacityRatioScorerFunction(scoringFunctionShape)}
+func RequestedToCapacityRatioResourceAllocationPriority(scoringFunctionShape FunctionShape, resources Resources) *ResourceAllocationPriority {
+	return &ResourceAllocationPriority{"RequestedToCapacityRatioResourceAllocationPriority", buildRequestedToCapacityRatioScorerFunction(scoringFunctionShape, resources), resources}
 }
 
-func buildRequestedToCapacityRatioScorerFunction(scoringFunctionShape FunctionShape) func(*schedulernodeinfo.Resource, *schedulernodeinfo.Resource, bool, int, int) int64 {
+func buildRequestedToCapacityRatioScorerFunction(scoringFunctionShape FunctionShape, resources Resources) func(map[v1.ResourceName]int64, map[v1.ResourceName]int64, bool, int, int) int64 {
```

Comment:
Could we still use `*schedulernodeinfo.Resource` instead of `map[v1.ResourceName]int64`, and build the map in this function? Then we do not need to change the interface `scorer func(requested, allocable *schedulernodeinfo.Resource, includeVolumes bool, requestedVolumes int, allocatableVolumes int) int64.
`

Reply:
requestToCapacityRatio is just an extension for the scorer function. The issue to address here was to allow users to able to incorporate extended resources in their scoring function. As most scoring functions use resource allocation codebase, i felt this was the best place to avoid code replication. Also, it would be useful for future variations for the scorer function.  All the dependent function calls have been modified and seem to work fine during testing.

Reply:
If I understand it correctly, `Resource.ScalarResources` is for extended resources.

Reply:
Yes 👍